app/controllers/catalog_controller.rb

```diff
@@ -743,48 +743,68 @@ def citation
+private
+
+  def render_empty_search
+    # This code is a copy of Blacklight::Catalog.index() method but adapted to use
+    # a cached version of the data rather than requesting the data from Solr.
+    # See https://github.com/projectblacklight/blacklight/blob/v7.0.1/app/controllers/concerns/blacklight/catalog.rb#L25-L41
+    @response = empty_solr_response
+    @document_list = @response.documents
+    respond_to do |format|
+      format.html { store_preferred_view }
+      format.rss  { render layout: false }
+      format.atom { render layout: false }
+      format.json do
+        @presenter = Blacklight::JsonPresenter.new(@response, blacklight_config)
       end
+      additional_response_formats(format)
+      document_export_formats(format)
     end
+  end
 
-    def home_page?
-      # When only the "controller" and "action" keys are in the request (i.e. no query or facets)
-      # we consider it the home page.
-      params.keys.count == 2
-    end
+  def home_page?
+    # When only the "controller" and "action" keys are in the request (i.e. no query or facets)
+    # we consider it the home page.
+    params.keys.count == 2
+  end
 
-    def empty_solr_response
-      raw_response = JSON.parse(empty_raw_response)
-      Blacklight::Solr::Response.new(raw_response, raw_response["responseHeader"]["params"], blacklight_config: @blacklight_config)
-    end
+  def empty_solr_response
+    raw_response = JSON.parse(empty_raw_response)
+    Blacklight::Solr::Response.new(raw_response, raw_response["responseHeader"]["params"], blacklight_config: @blacklight_config)
+  end
 
-    def empty_raw_response
-      Rails.cache.fetch("home_page_empty_raw_response", expires_in: 3.hours) do
-        Rails.logger.info "Cached home page results"
-        # We cannot cache the Blacklight::Solr::Response as-is so we convert it to JSON first
-        (response, _deprecated_document_list) = search_service.search_results
-        response.to_json
-      end
+  def empty_raw_response
+    Rails.cache.fetch("home_page_empty_raw_response", expires_in: 3.hours) do
+      Rails.logger.info "Cached home page results"
+      # We cannot cache the Blacklight::Solr::Response as-is so we convert it to JSON first
+      (response, _deprecated_document_list) = search_service.search_results
+      response.to_json
     end
+  end
 
-    def basic_response
-      render plain: 'OK'
-      nil
-    end
+  def basic_response
+    render plain: 'OK'
+    nil
+  end
+
+  def search_service_context
+    return {} unless Flipflop.multi_algorithm?
+    return {} unless alternate_search_builder_class # use default if none specified
+    { search_builder_class: alternate_search_builder_class }
+  end
+
+  def alternate_search_builder_class
+    return unless search_algorithm_param && allowed_search_algorithms.include?(search_algorithm_param)
+
+    "#{search_algorithm_param}_search_builder".camelize.constantize
+  end
+
+  def allowed_search_algorithms
```

Comment:
I think we should add some comment here about needing to add any new searchbuilder to here for algorithm switching
```suggestion
  # When adding a new Ranking Algorithm the name will need to be added to this list before it can be utilized
  # For example if you added a cats ranking algorithm, with a CatsSearchBuilder you would add "cats" to this list ["engineering","cats"]
  # This is to make sure the user can not just execute any SearchBuilder in the system
  def allowed_search_algorithms
```